scripts/dumps/bulk_genome_dump_from_core.py

```diff
@@ -0,0 +1,337 @@
+    sys.exit(1)
+
+
+def subprocess_call(
+    command,
+    stdout_file="/dev/null",
+    stderr_file="/dev/null",
+    work_dir=None,
+    shell=False,
+    use_job_scheduler=False,
+    job_name=None,
+):
+    """
+    Subprocess function to execute the given command line.
+
+    Args:
+        command (list): The command that the subprocess will execute.
+        work_dir (str): The location where the command should be run.
+        shell (bool): If True, the specified command will be executed through the shell.
+        use_job_scheduler (bool): If True, the command will be submitted to a job scheduler.
+
+    Returns:
+        str: The subprocess output or None otherwise.
+    """
+    if use_job_scheduler:
+        job_scheduler = detect_job_scheduler()
+
+        if job_scheduler == "NONE":
+            logging.error("No job scheduler detected.")
+            sys.exit(1)
+
+        if job_scheduler == "SLURM":
+            command = [
+                "sbatch",
+                "--time=1-00",
+                "--mem-per-cpu=4gb",
+                "--cpus-per-task=1",
+                "--export=ALL",
+                f"--output={stdout_file}",
+                f"--error={stderr_file}",
+                f"--job-name={job_name}",
+                f"--wrap={shlex.join(command)}",
+            ]
+        elif job_scheduler == "LSF":
+            command = [
+                "bsub",
+                "-W",
+                "1:00",
+                "-R",
+                "rusage[mem=4096]",
+                "-J",
+                job_name,
+                f"-o {stdout_file}",
+                f"-e {stderr_file}",
+            ] + command
+
+    logging.info("Running: %s", " ".join(command))
+    with subprocess.Popen(
+        command,
+        shell=shell,
+        cwd=work_dir,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        universal_newlines=True,
+    ) as process:
+        output, stderr = process.communicate()
+
+        if process.returncode != 0:
+            out = f"stdout={output}, stderr={stderr}"
+            logging.error(
+                "Command %s exited %d: %s", " ".join(command), process.returncode, out
+            )
+            raise RuntimeError(
+                f"Command {' '.join(command)} exited {process.returncode}: {out}"
+            )
+
+        logging.info("Successfully ran: %s", " ".join(command))
+        return output.strip()
+
+
+def download_file(
+    host,
+    port,
+    core_db,
+    fasta_filename,
+    stdout_file="/dev/null",
+    stderr_file="/dev/null",
+    genome_component="",
+    mask="soft",
+):
+    """
+    Download the FASTA file from the core DB using a PERL script `dump_genome_from_core.pl`.
+
+    Args:
+        host (str): The DB host name.
+        port (str): The port number.
+        core_db (str): The core_db name.
+        fasta_filename (str): The name given for the FASTA file.
+        mask (str): The mask format for the FASTA file.
```

Comment:
Remember to document all the parameters: 🙂 
```suggestion
        fasta_filename (str): The name given for the FASTA file.
        stdout_file (str): Job scheduler standard output file (default: '/dev/null').
        stderr_file (str): Job scheduler standard error file (default: '/dev/null').
        genome_component (str): Optional subgenome component of polyploid genome.
        mask (str): The mask format for the FASTA file.
```